### ISSUE TITLE
Support for private repositories

### ISSUE THREAD

**Issue by brunolemos** [devhubapp/devhub#32]
_If you want this feature for your **personal** private repositories, please react with a 👍._
_If you want this for your **team/company/organization** private repositories, react with a ❤️._
_Follow [@devhub_app](https://twitter.com/devhub_app) on Twitter to be notified when this becomes available._

### Add support for:

- Showing activities of private repositories
- Enhance notifications of private repositories

### Implementation requirements

- Request only the permissions that are strictly necessary
  - For this reason, we will need to migrate from `OAuth App` to `GitHub App` ([see comparison](https://developer.github.com/apps/differences-between-apps/))

#### Why not enable private access using the current implementation?
    
GitHub `OAuth Apps` have a very broad and dangerous permission scope. See this [petition](https://github.com/dear-github/dear-github/issues/113) for details. For example, to be able read activities (issue comments, ...) from private repositories, we need to request the `repo` scope. And this is what it gives access to:

<img height="250" alt="screen shot 2018-12-14 at 22 45 22" src="https://user-images.githubusercontent.com/619186/50036769-b9d96200-fff2-11e8-8ffb-4c6df6214682.png">

As a security precaution, DevHub does not intend to ask for this permission, since it doesn't ever need any code or write access.

---

TL/DR: This feature is planned and is high priority. 
But to do it right, first we need to make DevHub work as a `GitHub App` instead of `OAuth App`. This will cause an increase of server usage and costs.

_Follow [@devhub_app](https://twitter.com/devhub_app) on Twitter to be notified when this becomes available._

**Comment by brunolemos** [comment]
### Why not enable private access using the current implementation?

GitHub OAuth apps have a very broad and dangerous permission scope. See this [petition](https://github.com/dear-github/dear-github/issues/113) for details. For example, to be able read activities (issue comments, ...) from private repositories, we need to request the `repo` scope. And this is what it gives access to:

<img width="447" alt="screen shot 2018-12-14 at 22 45 22" src="https://user-images.githubusercontent.com/619186/50036769-b9d96200-fff2-11e8-8ffb-4c6df6214682.png">

For security reasons, DevHub does not intend to ask for this permission, since it doesn't ever need any code access. Specially write access. These tokens could become a target of hackers, so this is a precaution to avoid that and keep everything safe and sound.

**Comment by ifuller1** [comment]
> This will cause an increase of server usage and costs.
I'd pay some kind of licensing to use this. Pretty sure others would too, esp if it were on a per-user basis with a sensible cost.

**Comment by brunolemos** [comment]
@ifuller1 Thanks for your support! I hope more people think like you.

**Comment by grrowl** [comment]
Would be keen to kick in a few $ for server costs. Probably wouldn't subscribe per-user but would donate especially with transparency around costs on your end. Just my 2c.

**Comment by brunolemos** [comment]
@grrowl thanks!

**Comment by brunolemos** [comment]
Hello everyone, I wrote some important thoughts about the project sustainability, please join the discussion: https://github.com/devhubapp/devhub/issues/106

**Comment by edasque** [comment]
Testing this in 4.7.0 (Mac Electron client). I was able to add a repo (having had to install devhub for my org) and it worked. Pretty smooth. Does this work also for the organization tab. I only see activity on public there, no activity on the private repos (the settings have checks on private & public).

**Comment by ArondeParon** [comment]
Currently testing this in 0.47 as well, but even though I have installed the app, it is still telling me to "Install the Github App"

**Comment by edasque** [comment]
@Arondepardon did you authorize DevHub access to your organization?

**Comment by brunolemos** [comment]
Hello everyone, first version with support for private repositories is now available!
Please try it out and let me know if it worked well for you!

I'll tweet about it soon on [@devhub_app twitter](https://twitter.com/devhub_app).

**Comment by brunolemos** [comment]
@edasque thanks! I'll check org support. It should work.

@ArondeParon weird. can you try a logout/login? I'll try to reproduce.

**Comment by brunolemos** [comment]
@ArondeParon oh you tried via a private notification instead of a column, right?
Yeah if you logout/login it will work, it's missing a required token. I'll better handle this case.

**Comment by ArondeParon** [comment]
Thanks, working now!

**Comment by edasque** [comment]
@brunolemos let me know what I can do to help diagnose.

**Comment by brunolemos** [comment]
@edasque unfortunately it doesn't seem to be possible yet :(

The endpoint required to access private organization events is not available to GitHub Apps yet. See image below, it doesn't have the same icon as the public endpoint has:

![image](https://user-images.githubusercontent.com/619186/53466826-49a77d80-3a32-11e9-8bcc-81dd24e85ecb.png)

When I try to access this endpoint it returns an error `Resource not accessible by integration`.
We will need to wait for GitHub to implement this.

**Comment by brunolemos** [comment]
Hello everyone, 

Private repository support is now available for all platforms! (<a href="https://play.google.com/store/apps/details?id=com.devhubapp&utm_source=github_repo_link" target="_blank">Android</a>, <a href="https://itunes.apple.com/us/app/devhub-for-github/id1191864199?l=en&mt=8&utm_source=github_repo_link" target="_blank">iOS</a>, <a href="https://devhubapp.com/" target="_blank">Web</a>, <a href="https://github.com/devhubapp/devhub/releases" target="_self">Desktop</a>)
Please try it out and send feedbacks here.  💙
How could we make DevHub more useful to you and your team?


Just tweeted about it, in case you want to show some support:
https://twitter.com/devhub_app/status/1101138702749888515

**Comment by brunolemos** [comment]
@renosbest, @jerson, @rounakdatta, @luongvo, @gbougakov, @19majkel94, @nder, @Pizzacus, @alessandroconti97, @bhoomit, @zhigang1992, @cozyplanes, @drmaas, @pantharshit00, @wysow, @Somnath-Guthula, @kimroen, @lots0logs, @hpneo, @RandallKent, @jellea, @BaileyJM02, @gillchristian, @olpeh, @nchopra, @RickCogley, @yinx, @laughedelic, @jacksonisiah, @azertys, @vjFaLk, @bemayr, @i-mighty, @Integralist, @tuanmai, @KeziahMoselle, @thulio, @justhamade, @facumenzella, @invig, @brminnick, @jonathanvoelkle, @devshawn, @icarcal, @bordoni, @thomasklwong, @JAForbes, @mikemorris, @littleStudent, @FeliciousX

**Comment by brunolemos** [comment]
@pietrocaselani, @aydieneue, @Renerte, @ifuller1, @rbngzlv, @subhaminion, @achambers, @JCMais, @pongz79, @lardissone, @lucarge, @mittalyashu, @anurag-arjun, @Jatarii, @sirbrillig, @RoamingAnt, @kevinwmerritt, @akrawchyk, @mikesigs, @benweatherman, @Whitetigerswt, @lanzafame, @lullasy, @j33ty, @jgodoi, @steeve, @panva, @jonmillar, @watura, @zack9433, @amorist, @rogier, @TimMikeladze, @YannickDot, @mateusmedeiros, @id13, @GuillaumeRx, @neoromantic, @GFargo, @hstandaert, @rchinerman, @btimo, @philkeys, @bsubra, @nicolaschartiot, @Esteban-Rocha, @edasque, @Vinnie1991, @spotman, @bennetthardwick

**Comment by brunolemos** [comment]
@guyca, @mkhan117, @AhoyLemon, @MSF-Jarvis, @shaharmor, @maprules1000, @yasyf, @mauriciord, @emersonsiega, @lborie, @defts, @rlespinasse, @akhilnarang, @phibo23, @bilalbayasut, @renatoagds, @bgorriz, @mischnic, @hackweiser9553, @rexpan, @ametektcigeoff, @gregbacchus, @d1ngd0, @afrase, @graemeworthy, @DivineOmega, @jotamontecino, @quagliato, @pcooney10, @dmnlk, @gustavo-depaula, @louisguitton, @StefanoCecere, @wmaurer, @jovannypcg, @linuxjuggler, @gagan0723, @mattwoberts, @menjilx, @joshdholtz, @nesl247, @epicallan, @wearhere, @bradvogel, @michal-rumanek, @kernelp4nic, @itschap, @szhu, @michaelklopf, @timiwahalahti

**Comment by brunolemos** [comment]
@impressthenet, @ahollenbach, @Its-Alex, @Arjan-Zuidema, @vitordino, @deadcoder0904, @tikotzky, @jusx, @preschian, @fmfdias, @eunjae-lee, @fr-kyleli, @orther, @josebreijo, @miscer, @Snipey, @gen1321, @rorisme, @antonsamper, @keichinger, @tcK1, @vladimyr, @LeoColomb, @techniq, @ellord, @ashleyng, @bmeverett, @knownasilya, @tsub, @simonsmith, @bonty, @davidwilde, @mcaballeropinto, @dapids, @pocka, @sandiiarov, @deini, @LucasNaja, @therealparmesh, @valpinkman, @nicolasrouanne, @luciannojunior, @ArondeParon, @crosslife

**Comment by brunolemos** [comment]
@appledevhub
